syncopy/io/load_ft.py

```diff
@@ -0,0 +1,419 @@
+
+    FT     Syncopy
+
+    label - channel
+    trial - trial
+    time  - time
+
+    optional:
+    fsample - samplerate
+
+    cfg - X
+
+    '''
+    # for user info
+    struct_name = h5Group.name[1:]
+
+    # this should be fixed upstream such that
+    # the `defaultDimord` is indeed the default :)
+    AData = AnalogData(dimord=AnalogData._defaultDimord)
+
+    # probably better to define an abstract mapping
+    # if we want to support more FT formats in the future
+
+    # these are numpy arrays holding hdf5 object references
+    # e.i. one per trial, channel, time (per trial)
+    trl_refs = h5Group['trial'][:, 0]
+    time_refs = h5Group['time'][:, 0]
+    chan_refs = h5Group['label'][0, :]
+
+    if 'fsample' in h5Group:
+        AData.samplerate = h5Group['fsample'][0, 0]
+    else:
+        AData.samplerate = _infer_fsample(h5File[time_refs[0]])
+
+    # -- retrieve shape information --
+    nTrials = trl_refs.size
+
+    # peek in 1st trial to determine single trial shape
+    # we only support equal trial lengths at this stage
+    nSamples, nChannels = h5File[trl_refs[0]].shape
+    nTotalSamples = nTrials * nSamples
+
+    itemsize = h5File[trl_refs[0]].dtype.itemsize
+    # in Mbyte
+    trl_size = itemsize * nSamples * nChannels / 1e6
+
+    # assumption: single trial fits into RAM
+    if trl_size > 0.4 * mem_use:
+        msg = f"\nSingle trial is at least 40% of the requested chache size of {mem_use}MB\n"
+        msg += f"Still trying to load {trl_size:.1f}MB trials.."
+        SPYWarning(msg)
+
+    # -- IO process --
+
+    # create new hdf5 dataset for our AnalogData
+    # with the default dimord ['time', 'channel']
+    h5FileOut = h5py.File(AData.filename, mode="w")
+    ADset = h5FileOut.create_dataset("data",
+                                     dtype=np.float32,
+                                     shape=[nTotalSamples, nChannels])
+
+    pbar = tqdm(trl_refs, desc=f"{struct_name} - loading {nTrials} trials")
+    SampleCounter = 0   # trial stacking
+
+    # one swipe per trial
+    for tr in pbar:
+        ADset[SampleCounter:nSamples, :] = h5File[tr]
+        SampleCounter += nSamples
+    pbar.close()
+
+    AData.data = ADset
+
+    # -- trialdefinition --
+
+    nTr_rng = np.arange(nTrials)
+    sampleinfo = np.vstack([nTr_rng, nTr_rng + 1]).T * nSamples
+
+    offsets = []
+    # we need to look into the time vectors for each trial
+    for time_r in time_refs:
+        offsets.append(h5File[time_r][0, 0])
+    offsets = np.array(offsets)
+
+    trl_def = np.hstack([sampleinfo, offsets[:, None]])
+    AData.trialdefinition = trl_def
+
+    # each channel label is an integer array with shape (X, 1),
+    # where `X` is the number of ascii encoded characters
+    channels = [''.join(map(chr, h5File[cr][:, 0])) for cr in chan_refs]
+    AData.channel = channels
+
+    # -- additional fields --
+
+    # this is the most experimental part
+    AData.info = {}
```

Comment:
Now I get the `info` attribute. I like the idea! And yes, this definitely requires parsing the contents of the HDF5 group more thoroughly before attaching it to the instance.

Reply:
Nicely done!